File: leetcode/ds/961. N-Repeated Element in Size 2N Array/961. N-Repeated Element in Size 2N Array.cpp
class Solution {
    vector <int> cnt;
public:
    Solution() {
        cnt.resize(10001, 0);
    }
    // int repeatedNTimes(vector<int>& nums) {
        
    //     map <int,int> freqMap;
    //     int ans = 1;
    //     for(auto n : nums) {
    //       freqMap[n]++;
    //         if(freqMap[n] > 1) {ans = n; break;}
    //     }
   
    //     return ans;
    // }
    int repeatedNTimes(vector<int>& nums) {
        int ans;
        for(auto n : nums) {
            if(cnt[n]) {
                ans = n;
                break;
            }
            cnt[n]++;
        }
        return ans;
    }
};

File: leetcode/ds/961. N-Repeated Element in Size 2N Array/961. N-Repeated Element in Size 2N Array.rs
use std::collections::HashMap;
impl Solution {
    pub fn repeated_n_times(nums: Vec<i32>) -> i32 {
        let mut m = HashMap::new();
        let mut ans = 1;
        for n in nums {
            let mVal = m.get(&n);
            if let Some(_) = mVal {
                ans = n;
                break;
            } else {
                m.insert(n, 1);
            }
        }
        return ans;
    }
}